eventservice/api/controllers/events.py

```diff
@@ -650,8 +652,11 @@ def event_operation_permission_check(req_data=None, event_id=None):
             # If this is a group event, apply group authorization. Regular events can proceed like before.
             if not check_group_event_admin_access(event, group_memberships):
                 abort(401)
+            if req_data and set(req_data.get('groupIds')) != set(event.get('groupIds')):
```

Comment:
This check fails for events that do not contain the `groupIds` field. Please fix this and let me know after you have tested different scenarios. Thanks.

Reply:
I have some questions about it. events without groupIds should does not have 'createdByGroupId', then it is handled by the code if not event.get('createdByGroupId') and not event.get('groupIds'):

I am not sure it is right or not.

Reply:
This is for backward compatibility, i.e., current events with `createdByGroupId` and no `groupIds` in the data. Those events will not have `groupIds` and will throw a runtime error as the code tries to access the `groupIds` field in the above line, which doesn't exist. As these are optional fields, an event can contain either of these (associated with a single group vs. multiple groups) or both (groupIds should get priority in that case, I think), or none (events from sources like WebTools, which do not have any group associated with it).

Reply:
I see. shall we use or instead of and in that if condition?

Reply:
I updated the code after me and Sandeep discussed on it.

Reply:
Thanks.